identity/src/main/java/com/stripe/android/identity/ui/CameraScreenLaunchedEffectLight.kt

```diff
@@ -0,0 +1,106 @@
+package com.stripe.android.identity.ui
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.platform.LocalLifecycleOwner
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.NavController
+import com.stripe.android.identity.analytics.IdentityAnalyticsRequestFactory
+import com.stripe.android.identity.ml.FaceDetectorOutput
+import com.stripe.android.identity.ml.IDDetectorOutput
+import com.stripe.android.identity.navigation.CouldNotCaptureDestination
+import com.stripe.android.identity.navigation.navigateTo
+import com.stripe.android.identity.networking.models.VerificationPage
+import com.stripe.android.identity.states.IdentityScanState
+import com.stripe.android.identity.states.IdentityScanState.Companion.isBack
+import com.stripe.android.identity.states.IdentityScanState.Companion.isFront
+import com.stripe.android.identity.viewmodel.IdentityScanViewModel
+import com.stripe.android.identity.viewmodel.IdentityViewModel
+import kotlinx.coroutines.launch
+
+/**
+ * [CameraScreenLaunchedEffect] without checking pageAndModelFiles from IdentityViewModel.
+ */
+@Composable
+internal fun CameraScreenLaunchedEffectLight(
+    identityViewModel: IdentityViewModel,
+    identityScanViewModel: IdentityScanViewModel,
+    verificationPage: VerificationPage,
+    navController: NavController
+) {
+    val lifecycleOwner = LocalLifecycleOwner.current
+    LaunchedEffect(identityScanViewModel) {
+        // Handles interim result - track FPS
+        identityScanViewModel.interimResults.observe(lifecycleOwner) {
+            identityViewModel.fpsTracker.trackFrame()
+        }
+
+        // Handles final result - upload if success, transition to CouldNotCapture if timeout
+        identityScanViewModel.finalResult.observe(lifecycleOwner) { finalResult ->
+            lifecycleOwner.lifecycleScope.launch {
+                identityViewModel.fpsTracker.reportAndReset(
```

Comment:
looks like we're using launched effects (UI layer) to communicate viewModels (more no the controller layer).

This works but it might be hard to maintain long term. A potential alternative would be to have a coordinator that can be injected on both the parent and the child viewmodel, so the UI does not need to be involved. 

This is how we do this on Connections:

- Coordinator: https://github.com/stripe/stripe-android/blob/437c9c19977e5b9ccb7d3ddf11be49866f10543e/financial-connections/src/main/java/com/stripe/android/financialconnections/domain/NativeAuthFlowCoordinator.kt#L15
- Child sending a message: https://github.com/stripe/stripe-android/blob/728b2da18ba3a9541efabb86358f9e21e1ddccd4/financial-connections/src/main/java/com/stripe/android/financialconnections/features/manualentry/ManualEntryViewModel.kt#L152
- Parent viewmodel observing messages: https://github.com/stripe/stripe-android/blob/728b2da18ba3a9541efabb86358f9e21e1ddccd4/financial-connections/src/main/java/com/stripe/android/financialconnections/presentation/FinancialConnectionsSheetNativeViewModel.kt#L85

Reply:
Thanks for the pointers, this `LaunchedEffect` is a bit wonky as it's mostly inherited from the legacy architecture with Fragment and navigation components. 

The essential goal is to remove any `observe` and `collectLatest` and totally rely on IdentityScanViewModel's new `State` to drive UI, and handle these non-UI logics within `identityScanViewModel` itself without the need to reference the parent `IdentityViewModel`. 

- E.g we'll directly inject the `fpsTracker` into `IdentityScanViewModel` and handle the `trackFrame` within [IdentityScanViewModel.onInterimResult](https://github.com/stripe/stripe-android/blob/5a6ff9cfda1a0af74d567cd2f3918aca0306c1fb/identity/src/main/java/com/stripe/android/identity/viewmodel/CameraViewModel.kt#L101) callback.

Need to double check but I think with the decoupling there will be need to send messages from child`IdentityScanViewModel` to parent `IdentityViewModel`, but I have added your reference in the follow up ticket [here](https://jira.corp.stripe.com/browse/IDPROD-6555) incase we do need that!